src/twisted/trial/test/test_script.py

```diff
@@ -33,6 +41,99 @@ def sibpath(filename):
     return util.sibpath(__file__, filename)
 
 
+def logSomething() -> None:
+    """
+    Emit something to L{twisted.logger}.
+    """
+    Logger().info("something")
+
+
+def parseArguments(argv: List[str]) -> trial.Options:
+    """
+    Parse an argument list using trial's argument parser.
+    """
+    config = trial.Options()
+    config.parseOptions(argv)
+    return config
+
+
+def runFromConfig(config: trial.Options) -> trial.Options:
+    """
+    Run L{logSomething} as a test method using the given configuration.
+    """
+    runner = trial._makeRunner(config)
+    runner.stream = StringIO()
+    suite = TestSuite([pyunit.FunctionTestCase(logSomething)])
+    runner.run(suite)
+    return config
+
+
+runFromArguments = compose(runFromConfig, parseArguments)
```

Comment:
_Technically_ I suppose this is a function definition that needs a docstring.

Reply:
is there a meaningful way to give it a docstring?  I've seen people write things like:

```
x = ...
"""
Blah blah blah
"""
```

I know that Python itself doesn't do anything constructive with this but maybe one of the doc tools Twisted is using does?

Alternatively, maybe `compose` should synthesize a docstring based on its arguments? :slightly_smiling_face:

Reply:
I filed https://github.com/twisted/twisted/issues/11659